src/display.rs

```diff
@@ -163,11 +163,11 @@ impl Display {
             let height = cell_height as u32 * dimensions.lines_u32();
 
             let new_viewport_size = PhysicalSize::new(
-                (width + 2 * u32::from(config.padding().x)) as f64,
-                (height + 2 * u32::from(config.padding().y)) as f64);
+                (width + 2 * u32::from(config.padding().x) * dpr.floor() as u32) as f64,
+                (height + 2 * u32::from(config.padding().y) * dpr.floor() as u32) as f64);
```

Comment:
I feel like this might produce slightly incorrect results, why are you flooring the DPR here before finishing the calculation? Also the result itself seems to be `f64` so I'm not sure why everything's converted to whole numbers (even before the change you've made).

`width as f64 + 2. * config.padding().x * dpr` seems like the calculation we'd want to do here. And just in case the padding is converted to u32 (floored basically) because we need a padding that is smaller than the available pixels (we wouldn't want to round up a padding since it would reduce the terminal area), then something like this should be more appropriate:
`(width + 2 * u32::from(config.padding().x * dpr)) as f64`

Note that the latter version also is in line with the changes you've made in the renderer, I'd assume this might just have been accidental? If there is a reason for this, please let me know.

Reply:
Totally an accident after many iterations of going back and forth trying to get it to work.